src/constraint_system/boolean.rs

```diff
@@ -57,27 +57,32 @@ where
     pub fn is_zero_gate(&mut self, a: Variable) -> Variable {
         //Get relevant field values
         let a_value = self.variables.get(&a).unwrap();
-        let a_inv_value = a_value.inverse()
-            .unwrap_or(E::Fr::one());
+        let a_inv_value = a_value.inverse().unwrap_or(E::Fr::one());
         //This has value 1 if input value is zero, value 0 otherwise
         let result_value = E::Fr::one() - *a_value * a_inv_value;
 
         let a_inv = self.add_input(a_inv_value);
         let result = self.add_input(result_value);
 
         // Enforce constraints
-        let a_times_result = self.arithmetic_gate(|gate| gate.witness(a, result, None).mul(E::Fr::one()));
+        let a_times_result = self.arithmetic_gate(|gate| {
+            gate.witness(a, result, None).mul(E::Fr::one())
+        });
         self.assert_equal(a_times_result, self.zero_var());
 
-        let a_times_ainv = self.arithmetic_gate(|gate| gate.witness(a, a_inv, None).mul(E::Fr::one()));
-        let sum = self.arithmetic_gate(|gate| gate.witness(a_times_ainv, result, None).add(E::Fr::one(), E::Fr::one()));
+        let a_times_ainv = self.arithmetic_gate(|gate| {
+            gate.witness(a, a_inv, None).mul(E::Fr::one())
+        });
+        let sum = self.arithmetic_gate(|gate| {
+            gate.witness(a_times_ainv, result, None)
+                .add(E::Fr::one(), E::Fr::one())
+        });
         let one = self.add_input(E::Fr::one());
         self.assert_equal(sum, one);
```

Comment:
The `assert_equal` function ought to be used to compare variables, rather than constrain the output of a gate to a constant. Usually you can accomplish the same thing in the gate itself by using the constant wire. See here: https://github.com/ZK-Garage/plonk/blob/8080e7dd6d31b5fe4ff04e7c8f5bb69c21eeb09c/src/constraint_system/arithmetic.rs#L74